llvm/lib/Transforms/Vectorize/SandboxVectorizer/Passes/RegionsFromMetadata.cpp

```diff
@@ -0,0 +1,30 @@
+//===- RegionsFromMetadata.cpp - A helper to test RegionPasses -----------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "llvm/Transforms/Vectorize/SandboxVectorizer/Passes/RegionsFromMetadata.h"
+
+#include "llvm/SandboxIR/Region.h"
+#include "llvm/Transforms/Vectorize/SandboxVectorizer/SandboxVectorizerPassBuilder.h"
+
+namespace llvm::sandboxir {
+
+RegionsFromMetadata::RegionsFromMetadata(StringRef Pipeline)
+    : FunctionPass("regions-from-metadata"), RPM("rpm") {
+  RPM.setPassPipeline(Pipeline, SandboxVectorizerPassBuilder::createRegionPass);
```

Comment:
Would it be possible to set the pipeline in RPM's constructor? rather than having to call `setPassPipeline` ?

Reply:
It's possible. Added a `PassManager(StringRef Name, StringRef Pipeline, CreatePassFunc CreatePass)` constructor to `PassManager` and similar ones to `FunctionPassManager` and `RegionPassManager`. Now both RegionsFromMetadata and BottomUpVec initialize the pipeline using the new RPM constructor

What's the advantage you see in doing so?

Reply:
It's just that it forces you to initialize the pipeline when you define a pass.